Compiled linux amd64 binary cant find resources

Hi

I compiled this project using:

```
GOOS=linux GOARCH=amd64 go build # In the projects directory
```

After uploading it to my `ubuntu 16.04` instance, i tried starting it using:

```
 /opt/shorturl/shorturl -baseurl http://localhost:8000
```

This resulted in the following error message:

```
panic: could not locate box "templates"

goroutine 1 [running]:
github.com/GeertJohan/go%2erice.MustFindBox(0x832bd5, 0x9, 0xc42000f9b0)
        /Users/vncntvandriessche/Documents/go/src/github.com/GeertJohan/go.rice/box.go:110 +0x97
main.NewServer(0x833e0d, 0xc, 0x7ffc7ae7aed8, 0x15, 0xc4201001e0)
        /Users/vncntvandriessche/Documents/go/src/github.com/shorturl/server.go:235 +0x38e
main.main()
        /Users/vncntvandriessche/Documents/go/src/github.com/shorturl/main.go:39 +0x2d7
```

I tried moving the resources folders into the same directory but to no avail.

--- a/.gitignore
+++ b/.gitignore
@@ -1,2 +1,3 @@
 *.db
 shorturl
+rice-box.go

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,19 +1,30 @@
-FROM golang:alpine
+# Build
+FROM golang:alpine AS build
 
-EXPOSE 8000/tcp
+ARG TAG
+ARG BUILD
 
-ENTRYPOINT ["shorturl"]
+ENV APP shorturl
+ENV REPO prologic/$APP
 
-RUN \
-    apk add --update git && \
+RUN apk add --update git make build-base && \
     rm -rf /var/cache/apk/*
 
-RUN mkdir -p /go/src/shorturl
-WORKDIR /go/src/shorturl
+WORKDIR /go/src/github.com/$REPO
+COPY . /go/src/github.com/$REPO
+RUN make TAG=$TAG BUILD=$BUILD build
 
-COPY . /go/src/shorturl
+# Runtime
+FROM scratch
 
-RUN go get -v -d
-RUN go get github.com/GeertJohan/go.rice/rice
-RUN rice embed-go
-RUN go install -v
+ENV APP shorturl
+ENV REPO prologic/$APP
+
+LABEL shorturl.app main
+
+COPY --from=build /go/src/github.com/${REPO}/${APP} /${APP}
+
+EXPOSE 8080/tcp
+
+ENTRYPOINT ["/shorturl"]
+CMD []

--- a/Makefile
+++ b/Makefile
@@ -1,16 +1,34 @@
-.PHONY: dev build clean
+.PHONY: dev build image test deps clean
+
+CGO_ENABLED=0
+COMMIT=`git rev-parse --short HEAD`
+APP=shorturl
+REPO?=prologic/$(APP)
+TAG?=latest
+BUILD?=-dev
 
 all: dev
 
 dev: build
-	./shorturl
+	@./$(APP)
 
-build: clean
-	go get ./...
-	go build .
+deps:
+	#@go get github.com/GeertJohan/go.rice/rice
+	@go get ./...
+	#@rice embed-go
+
+build: clean deps
+	@echo " -> Building $(TAG)$(BUILD)"
+	@go build -tags "netgo static_build" -installsuffix netgo \
+		-ldflags "-w -X github.com/$(REPO).GitCommit=$(COMMIT) -X github.com/$(REPO).Build=$(BUILD)" .
+	@echo "Built $$(./$(APP) -v)"
+
+image:
+	@docker build --build-arg TAG=$(TAG) --build-arg BUILD=$(BUILD) -t $(REPO):$(TAG) .
+	@echo "Image created: $(REPO):$(TAG)"
 
 test:
-	go test ./...
+	@go test -v -cover -race $(TEST_ARGS)
 
 clean:
-	rm -rf shorturl
+	@rm -rf $(APP)

--- a/main.go
+++ b/main.go
@@ -1,5 +1,7 @@
 package main
 
+//go:generate rice embed-go
+
 import (
 	"log"
 

